### DISCUSSION TITLE
PIXIJS Sort/Anchor 0.5

### DISCUSSION THREAD

**Discussion by UnDVD2Euros** [pixijs/pixijs#8143] [General]
Hello, quick question !

this.YSort = [];
this one contains each object on the stage including the player every object anchor are fixed at 0.5 (center center)

and here i try to sort them on stage
this.YSort.children.sort((a, b) => {
      return a.y - b.y;
});

i think the problem is probably cause object are centered cause before i swap my engine i was anchoring at 0.5, 1 (center bottom) and it was working properly.

is there a way to return something that can verify even if i'm using anchor(0.5);

**Comment by bigtimebuddy** [comment]
Maybe try this:

```js
this.YSort.children.sort((a, b) => {
return (a.y + (a.height/2)) - (b.y + (b.height/2);
});
```
We sort by the bottom of the image (assuming anchor is in the middle).

**Reply by UnDVD2Euros** [reply]
Genius i did something like that but i just missed the division by 2, thanks.

**Reply by UnDVD2Euros** [reply]
But now it's not working on every object on the map, just a couple of kinda weird
